Initialize token implementation

Namaskar-1F64F opened the pull request:
This PR addresses the comment that anyone can call initialize on the bond contract. I think this makes it look bad and do not think there is a reason to do this besides the implementation contract's bond could be initialized with some advertisement or other unsavory bond name and symbol. This would be seen when our bond is verified on etherscan I believe. The bond will not be marked as a "clone" of the factory because it was not created through the `createBond` method.

RusseII commented:
> This PR addresses the comment that anyone can call initialize on the bond contract. I think this makes it look bad and do not think there is a reason to do this besides the implementation contract's bond could be initialized with some advertisement or other unsavory bond name and symbol

Do we think we should merge it?

Namaskar-1F64F commented:
> > This PR addresses the comment that anyone can call initialize on the bond contract. I think this makes it look bad and do not think there is a reason to do this besides the implementation contract's bond could be initialized with some advertisement or other unsavory bond name and symbol
> 
> Do we think we should merge it?

I'm on the fence. I think no, but we should consider initializing it ourselves after it's deployed.
